cardano-cli/test/Test/Cli/MultiAssetParsing.hs

```diff
@@ -3,171 +3,41 @@
 
 module Test.Cli.MultiAssetParsing where
 
-import           Cardano.Prelude
+import           Cardano.Prelude hiding (filter)
 
-import           Cardano.Api.Typed
-import           Cardano.CLI.Mary.Parser
 import qualified Data.Text as Text
-import           Test.Cli.Gen
-import           Text.Parsec (ParseError)
 import qualified Text.Parsec as Parsec (parse)
-import           Text.Parsec.String (Parser)
 
-import           Hedgehog (Gen, Property, checkSequential, discover, evalEither, forAll, property,
-                     (===))
-import           Hedgehog.Internal.Property (failWith)
+import           Hedgehog (Property, checkSequential, discover, forAll, property, tripping)
+import           Hedgehog.Gen (filter)
 
-{- HLINT ignore "Reduce duplication" -}
+import           Cardano.Api.Typed (valueToList)
+import           Cardano.CLI.Mary.RenderValue (defaultRenderPrettyValueOptions,
+                     defaultRenderValueOptions, renderPrettyValue, renderValue)
+import           Cardano.CLI.Mary.ValueParser (parseValue)
 
--- Lexer
-lex :: Parser a -> Text -> Either ParseError a
-lex p = Parsec.parse p "" . Text.unpack
+import           Test.Cardano.Api.Typed.Gen (genValueDefault)
 
--- Parser
-parse :: TParser a -> Tokens -> Either ParseError a
-parse p = Parsec.parse p ""
-
--- Lexing
-
-prop_lexLovelace :: Property
-prop_lexLovelace =
-  property $ do
-    (input, expectedOutput) <- forAll genLovelaceToken
-    case lex lexToken input of
-      Left pe -> failWith Nothing $ show pe
-      Right token -> token === expectedOutput
-
-prop_lexValue_fullySpecified :: Property
-prop_lexValue_fullySpecified =
+prop_roundtrip_Value_parse_render :: Property
+prop_roundtrip_Value_parse_render =
   property $ do
-    (input, expectedOutput) <- forAll genValueTokenFullySpecified
-    case lex valueTokenFullySpecified input of
-      Left pe -> failWith Nothing $ show pe
-      Right token -> token === expectedOutput
+    value <- forAll $ filter (not . null . valueToList) genValueDefault
```

Comment:
Ah, actually maybe I should support passing a `Range` to the generator. Either that or we could support parsing an empty string as an empty `Value`, but I'm not sure that's appropriate.